engine/grid.py

```diff
@@ -301,79 +320,379 @@ def find_border_nodes(self):
+            return
+        # Column where we want to begin turning, one node closer to the inside of our shape
+        turning_row = self.get_turning_column(edge_row) 
+        next_column = self.curr_pos[0] + 1
+        # Traverse the row until it is time to turn
+        while not self.is_before_turning_column(turning_row):
+            self.waypoints.append(new_pos)
+            self.curr_pos = new_pos
+            new_pos = self.get_next_traversal_pos()
+
+        # Check whether turning to the next row is valid
+        if not next_column < self.num_rows or not self.nodes[next_column][turning_row].is_active:
+            # Next row is out of bounds of the grid OR the node in the row above,
+            # where we are trying to turn to, isn't active for traversal
+            self.waypoints_is_finished = True
+        else:
+            # Turn to next row
+            # Create waypoints needed for a smooth turning trajectory to 
+            # "guide" our robot to the next original waypoint
+            circle_plt = self.plot_circle((self.curr_pos[0], turning_row), (next_column, turning_row),
+                                        (self.curr_pos[0] + .5, turning_row), self.get_turn_orientation())                                       
+            self.waypoints += circle_plt
+            # Update traversal details
+            self.switch_directions()
+            # self.curr_pos = (turning_column, next_row)
+            self.curr_pos = (next_column, turning_row)
+
+    ##Given border and active nodes, compute lawnmower traversal
+    def get_all_guided_lawnmower_waypoints_adjustable(self, is_vertical):
+        """
+        Returns the waypoints being traversed. The algorithm traverses from bottom
+        up starting from left to right.
+
+        There are two big branches:
+            * Direction.LEFT branch handles traversal from right towards left and
+            turning clock wise at the end of the row (or terminating).
+            * Direction.RIGHT branch handles traversal from left towards right
+            and turning counter clock wise at the end of the row (or terminating).
+        The plot_circle plots the circular waypoints during the turn
+        """
+        # TODO: move find border nodes into this func instead of doing it
+        # everytime before this is called
+        self.curr_pos = self.leftmost_node_pos
+        self.waypoints.append(self.curr_pos)
+        if is_vertical:
+            self.direction = self.direction.UP
+        while not self.waypoints_is_finished:
+            if self.curr_pos == None:
+                self.waypoints_is_finished = True
+
+            # Question: can we delete? bounds checked in add guided? change to assert?
+            # elif (self.curr_pos[0] > self.num_rows - 1 or 
+            #     self.curr_pos[0] < 0 or self.curr_pos[1] > self.num_cols - 1 or 
+            #     self.curr_pos[1] < 0):
+            #     self.phase = self.WaypointPhase.TERMINATE
```

Comment:
```suggestion
```
Do uk if we still need this? If so, let's mention it in pr summary/other. If not, suggested deletion.

Reply:
I don't think we need this, I deleted them